spec/controllers/irs_attempts_api_controller_spec.rb

```diff
@@ -0,0 +1,111 @@
+require 'rails_helper'
+
+RSpec.describe IrsAttemptsApiController do
+  before do
+    stub_analytics
+
+    IrsAttemptsApi::RedisClient.clear_attempts!
+    existing_events
+
+    request.headers['Authorization'] = "Bearer #{auth_token}"
+  end
+
+  let(:auth_token) do
+    IdentityConfig.store.irs_attempt_api_auth_tokens.split(',').first
+  end
+  let(:existing_events) do
+    3.times.map do
+      jti, jwe = IrsAttemptsApi::EncryptedEventTokenBuilder.new(
+        event_type: :test_event,
+        session_id: 'test-session-id',
+        occurred_at: Time.zone.now,
+        event_metadata: {},
+      ).build_event_token
+      IrsAttemptsApi::RedisClient.new.write_event(jti: jti, jwe: jwe)
+      [jti, jwe]
+    end
+  end
+  let(:existing_event_jtis) { existing_events.map(&:first) }
+
+  describe '#create' do
+    it 'authenticates the client' do
+      request.headers['Authorization'] = auth_token # Missing Bearer prefix
+
+      post :create, params: {}
+
+      expect(response.status).to eq(401)
+
+      request.headers['Authorization'] = 'garbage-fake-token-nobody-likes'
+
+      post :create, params: {}
+
+      expect(response.status).to eq(401)
+    end
+
+    it 'allows events to be acknowledged' do
+      expect(@analytics).to receive(:irs_attempts_api_events).with(
+        acknowledged_event_count: 3,
+        rendered_event_count: 0,
+        set_errors: nil,
+      )
```

Comment:
no need to change, this is just me thinking aloud... as we've migrated other events to the new typed AnalyticsEvents, we've been checking for strings ex 

```
expect().to receive(:track_event).with('The String Key', value: 1)
```

or the

```
expect(analytics).to have_logged_event('string key', value: 1)
```

but maybe this works just as well? Do we have a preference we should aim for?

Reply:
I forgot about `have_logged_event` which I think I actually like a little bit better.